### PR TITLE
Lots of updates from other pipelines

### PR DESCRIPTION
**Pull request by ewels**
* Added Singularity example file
* Updated config profiles
* Refactored testing with -profile test
* Fixed errors with email software_versions

### CODE REVIEW

File: {{cookiecutter.pipeline_slug}}/conf/base.config [hunk 1]
```diff
@@ -36,4 +36,5 @@ params {
   max_memory = 128.GB
   max_cpus = 16
   max_time = 240.h
+  igenomes_base = 's3://ngi-igenomes/igenomes/'
```
Comment (apeltzer) [inline {{cookiecutter.pipeline_slug}}/conf/base.config:39]: Does that not have an impact on people without s3 access?

**Review by apeltzer — APPROVED** [review]
Looks all good to me!

File: {{cookiecutter.pipeline_slug}}/conf/base.config (hunk 1 above)
Comment (ewels) [reply {{cookiecutter.pipeline_slug}}/conf/base.config:39]: No - it's public so as long as you have an internet connection, nextflow should be able to download the refs I think.

Comment (apeltzer) [reply {{cookiecutter.pipeline_slug}}/conf/base.config:39]: This is cool then - thanks for clarifying!

**Pull request merged**